sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Implicits.scala

```diff
@@ -62,4 +86,50 @@ object DataSourceV2Implicits {
       new CaseInsensitiveStringMap(options.asJava)
     }
   }
+
+  implicit class PartitionSpecsHelper(partSpecs: Seq[PartitionSpec]) {
+    def asUnresolvedPartitionSpecs: Seq[UnresolvedPartitionSpec] =
+      partSpecs.map(_.asInstanceOf[UnresolvedPartitionSpec])
+
+    def asResolvedPartitionSpecs: Seq[ResolvedPartitionSpec] =
+      partSpecs.map(_.asInstanceOf[ResolvedPartitionSpec])
+  }
+
+  implicit class TablePartitionSpecHelper(partSpec: TablePartitionSpec) {
+    def asPartitionIdentifier(partSchema: StructType): InternalRow = {
```

Comment:
let's move it to `ResolvePartitionSpec` and turn it into a method not implicit class.